File: src/main/java/com/example/blog/entity/ImageEntity.java
package com.example.blog.entity;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotEmpty;
import java.time.LocalDateTime;

//todo: remove setter
@Entity
@Getter
@Setter
@NoArgsConstructor
public class ImageEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotEmpty
    private String originalName;
    @NotEmpty
    private String filePath;
    @CreationTimestamp
    private LocalDateTime createAt;
    @UpdateTimestamp
    private LocalDateTime UpdatedAt;

    public ImageEntity(String originalName, String filePath) {
        this.originalName = originalName;
        this.filePath = filePath;
    }
}

File: src/main/kotlin/com/example/blog/entity/ImageEntity.kt
package com.example.blog.entity

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.nio.file.Path
import java.nio.file.Paths
import java.time.LocalDateTime
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
class ImageEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long = 0,

    private val originalName: String,

    private val filePath: String,

    @CreationTimestamp
    private val createAt: LocalDateTime = LocalDateTime.now(),

    @UpdateTimestamp
    private val UpdatedAt: LocalDateTime = LocalDateTime.now()
) {
    fun getPath(): Path {
        return Paths.get(filePath)
    }
}
